src/schema/userExperience.ts

```diff
@@ -1,36 +1,56 @@
 import { Raw } from 'typeorm';
-import { z } from 'zod';
 import { IResolvers } from '@graphql-tools/utils';
 import { UserSkill } from '../entity/user/UserSkill';
-import { Company, CompanyType } from '../entity/Company';
-import { BaseContext, Context } from '../Context';
+import { Company } from '../entity/Company';
+import { AuthContext, BaseContext } from '../Context';
 import { traceResolvers } from './trace';
-import { ExperienceStatus } from '../entity/user/experiences/types';
+import {
+  ExperienceStatus,
+  ProjectLinkType,
+  UserExperienceType,
+  WorkEmploymentType,
+  WorkVerificationStatus,
+} from '../entity/user/experiences/types';
 import { queryReadReplica } from '../common/queryReadReplica';
-import { ValidationError } from 'apollo-server-errors';
 import { UserExperience } from '../entity/user/experiences/UserExperience';
-
-export enum ExperienceAutocompleteType {
-  JobTitle = 'job_title',
-  CertificationName = 'certification_name',
-  AwardName = 'award_name',
-  AwardIssuer = 'award_issuer',
-  PublicationPublisher = 'publication_publisher',
-  CourseInstitution = 'course_institution',
-  FieldOfStudy = 'field_of_study',
-}
-
-const experiencePropertyByType: Record<ExperienceAutocompleteType, string> = {
-  [ExperienceAutocompleteType.JobTitle]: 'title',
-  [ExperienceAutocompleteType.CertificationName]: 'title',
-  [ExperienceAutocompleteType.AwardName]: 'title',
-  [ExperienceAutocompleteType.AwardIssuer]: 'issuer',
-  [ExperienceAutocompleteType.PublicationPublisher]: 'publisher',
-  [ExperienceAutocompleteType.CourseInstitution]: 'institution',
-  [ExperienceAutocompleteType.FieldOfStudy]: 'fieldOfStudy',
-} as const;
```

Comment:
There are lot of changes here, the ones outside the resolver are mainly shared code moved to `'../common/userExperience'`.